forc-plugins/forc-client/src/util/tx.rs

```diff
@@ -121,13 +129,65 @@ impl<Tx: Buildable + SerializableVec + field::Witnesses + Send> TransactionBuild
         client: FuelClient,
         unsigned: bool,
         signing_key: Option<SecretKey>,
+        wallet_mode: WalletSelectionMode,
     ) -> Result<Tx> {
         let params = client.chain_info().await?.consensus_parameters.into();
         let mut signature_witness_index = 0u8;
-        if !unsigned {
+        let signing_key = if !unsigned {
+            let key = match (wallet_mode, signing_key) {
+                (WalletSelectionMode::ForcWallet, None) => {
+                    // TODO: This is a very simple TUI, we should consider adding a nice TUI
+                    // capabilities for selections and answer collection.
+                    let wallet_path = default_wallet_path();
+                    if !wallet_path.exists() {
+                        anyhow::bail!("Cannot find a wallet at {wallet_path:?}\nPlease a generate new wallet with `forc wallet new`");
```

Comment:
Perhaps here instead of bailing, we could prompt the user is they would like a new wallet generated for them. If they select no then we can return at that point.

Reply:
It sounds good! but I have to expose that part as well. For the improved UX PR that I will have next, I have couple of other things to expose as well. Such as listing accounts so that users can select which account to sign with and balance queries so that we can offer them to go to faucet. I can add expose creating a new wallet alongside them or if you think we should have it with this PR, we will have to wait for forc-wallet changes first.

Reply:
That's fine we can do that in a follow up PR. Maybe just open an issue so we don't forget.